src/Voron/StorageEnvironmentOptions.cs

```diff
@@ -29,7 +29,22 @@ public abstract class StorageEnvironmentOptions : IDisposable
         public event EventHandler<NonDurabalitySupportEventArgs> OnNonDurabaleFileSystemError;
         private long _reuseCounter;
         public abstract override string ToString();
-        public bool ForceUsing32BitsPager { get; set; }
+
+        private bool _forceUsing32BitsPager;
+        public bool ForceUsing32BitsPager
+        {
+            get
+            {
+                return _forceUsing32BitsPager;
+            }
+            set
+            {
+                _forceUsing32BitsPager = value;
+                MaxLogFileSize = (value ? 32 : 256)*Constants.Size.Megabyte;
```

Comment:
This is valid only if the user didn't already change it from the default values.